lib/spack/spack/spec.py

```diff
@@ -802,7 +802,7 @@ def _libs_default_handler(descriptor, spec, cls):
     Raises:
         RuntimeError: If no libraries are found
     """
-    name = 'lib' + spec.name
+    name = 'lib' + spec.name.replace('-', '?')
```

Comment:
I might not be getting something trivial, but I'll go ahead and ask anyway. I assume `spec.name` is a string. [In the docs](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str) for python `str` they say:

> str.replace(old, new[, count])
    Return a copy of the string with all occurrences of substring old replaced by new. If the optional argument count is 
    given, only the first count occurrences are replaced.

Is this the behavior we want? I mean:
```python
>>> 'foo-bar'.replace('-', '?')
'foo?bar'
```